### ISSUE TITLE
When debugging typescript attached to Chrome, compilation opens editor tabs with compiled code 

### ISSUE THREAD

**Issue by thomas-klambauer** [Microsoft/vscode-chrome-debug#717]
VSCode Version: 1.25.1
OS Version: windows 10
Steps to Reproduce:

Debug a typescript project, start debugging with:
ng serve --proxy-config proxy.conf.json --base-href /
and config:
"type": "chrome",
"request": "launch",
and
"type": "chrome",
"request": "attach",
Change some typescript, compile runs in background and opens tabs, eg "main.bundle.js, styles.bundle.js which seem to be compiled files. this also steals focus.

**Comment by thomas-klambauer** [comment]
![43627899-8bc68b26-96f8-11e8-86b6-278163303a21](https://user-images.githubusercontent.com/12692832/43629999-3e383a38-96ff-11e8-8007-13b635083618.png)

**Comment by roblourens** [comment]
Are you hitting a breakpoint, or does the editor just open files by itself? And why are they empty?

**Comment by thomas-klambauer** [comment]
It appears empty during compilation I think. When it's done, the content is there alright:
![withcode](https://user-images.githubusercontent.com/12692832/43703511-3d64de08-995d-11e8-93db-99bd66c0570f.png)

The tab pops up each time when a file is changed + saved (html or ts). Not related to a breakpoint, so debugging without changing the files does not cause the issue.

the popping up of tabs is delayed if the debugged browser window shows an alert message.

So the behavior is:
* Start debugging with an attached chrome process
* During debugging change file x.ts + save
* Upon save, it seems the compile starts in the backgroup (tsc?), then it opens the number of tabs with the compiled content (in "preview mode", so only one tab opens, but multiple files are loaded in that).

**Comment by roblourens** [comment]
Ok, so this file opens by itself, without you opening it yourself before?

And it only happens while debugging? If you start the build task, and change some files, without being in debug mode, do the files ever open?

Can you set `"trace": true` in your launch config, run this again, and attach the log file here?

**Comment by thomas-klambauer** [comment]
Current relevant launch config entries for reference:
`        {
            "name": "Launch Chrome",
            "type": "chrome",
            "request": "launch",
            "url": "http://localhost:4200/#",
            "webRoot": "${workspaceFolder}"
        },
        {
            "name": "Attach Chrome",
            "type": "chrome",
            "request": "attach",
            "port": 9222,
            "webRoot": "${workspaceFolder}"
        },
    ],
    "compounds": [
        {
            "name": "Launch + debug chrome",
            "configurations": ["Launch Chrome", "Attach Chrome"]
        }
    ]    `

I am starting the compound ""Launch + debug chrome""

**Comment by thomas-klambauer** [comment]
Some more details: 
* it also already opens the files when I start debugging (plus every time after that on re-compilation)

At first I was not able to reproduce the issue, but after a pc-restart it was there again (maybe a sporadic issue?).

Note, I added the "trace": true to _both_ configs in my compound config - both processes seem to write to the same file, let me know if that corrupts the file sor something.

Attached it here - zipped.
[vscode-chrome-debug.zip](https://github.com/Microsoft/vscode-chrome-debug/files/2265274/vscode-chrome-debug.zip)

**Comment by thomas-klambauer** [comment]
> If you start the build task, and change some files, without being in debug mode, do the files ever open?

No, I had "ng serve" running in the console, but not the debugger: changing does a recompile, but no tabs openend.

Then when starting the debugging, the issue does occur.

**Comment by thomas-klambauer** [comment]
Another interesting observation. I killed the debugged chrome during that compilation and got this error in the tab that opened. Maybe that helps?

![2018-08-07 08_46_44-window](https://user-images.githubusercontent.com/12692832/43759070-6fdf93a2-9a1e-11e8-8f5c-2184d1da7a29.png)

**Comment by roblourens** [comment]
Any idea @isidorn?

**Comment by roblourens** [comment]
Wait, why are you running two debug configs against the same tab? Does it happen if you don't do that?

**Comment by thomas-klambauer** [comment]
Hmm what do you mean exactly? Do you mean the compound launch config? I was under the impression that I need a "launch" and "attach" request - actually just an "attach" won't start chrome.

**Comment by roblourens** [comment]
"launch" will launch chrome and attach to it so you can debug. "attach" is to attach to an already running instance of chrome.

I'm guessing in this case you've attached twice to the same tab.
